tests/bgp/test_ipv6_bgp_scale.py

```diff
@@ -0,0 +1,376 @@
+        if 'ipv6' in tbinfo['topo']['properties']['configuration'][hostname]['interfaces']['Ethernet1']:
+            bgp_info[hostname][IPV6_KEY] = \
+                tbinfo['topo']['properties']['configuration'][hostname]['interfaces']['Ethernet1']['ipv6'].split('/')[0]
+        elif 'lacp' in tbinfo['topo']['properties']['configuration'][hostname]['interfaces']['Ethernet1']:
+            pc_name = 'Port-Channel' + \
+                str(tbinfo['topo']['properties']['configuration'][hostname]['interfaces']['Ethernet1']['lacp'])
+            bgp_info[hostname][IPV6_KEY] = \
+                tbinfo['topo']['properties']['configuration'][hostname]['interfaces'][pc_name]['ipv6'].split('/')[0]
+    logging.info("BGP peers info: %s", bgp_info)
+    return bgp_info
+
+
+@pytest.fixture(scope="function")
+def announce_bgp_routes_teardown(localhost, tbinfo):
+    yield
+    topo_name = tbinfo['topo']['name']
+    ptf_ip = tbinfo['ptf_ip']
+    localhost.announce_routes(
+        topo_name=topo_name,
+        ptf_ip=ptf_ip,
+        action="announce",
+        path="../ansible/",
+        log_path="logs"
+    )
+
+
+def get_all_bgp_ipv6_routes(duthost):
+    return json.loads(
+        duthost.shell("docker exec bgp vtysh -c 'show ipv6 route bgp json'")['stdout']
+    )
+
+
+def generate_packets(routes, dut_mac, src_mac):
+    pkts = []
+    for prefix in routes.keys():
+        addr = str(ipaddress.ip_network(prefix)[1])
+        pkt = simple_icmpv6_packet(
+            eth_dst=dut_mac,
+            eth_src=src_mac,
+            ipv6_dst=addr
+        )
+        pkts.append(bytes(pkt))
+
+    return pkts
+
+
+def change_routes_on_peers(localhost, topo_name, ptf_ip, routes, action, peers):
+    localhost.announce_routes(
+        topo_name=topo_name,
+        adhoc=True,
+        ptf_ip=ptf_ip,
+        action=action,
+        routes=routes,
+        peers=peers,
+        path="../ansible/",
+        log_path="logs"
+    )
+
+
+def remove_nexthops_in_routes(routes, nexthops):
+    ret_routes = dict(routes)  # make a deep copy here
+    prefxies_to_remove = []
+    for prefix, attr in ret_routes.items():
+        _nhs = [nh for nh in attr[0]['nexthops'] if nh['ip'] not in nexthops]
+        if len(_nhs) == 0:
+            prefxies_to_remove.append(prefix)
+        else:
+            attr[0]['nexthops'] = _nhs
+    for prefix in prefxies_to_remove:
+        ret_routes.pop(prefix)
+    return ret_routes
+
+
+def compare_routes(expected_routes, running_routes):
+    if len(expected_routes) != len(running_routes):
+        return False
+    for prefix, attr in expected_routes.items():
+        if prefix not in running_routes:
+            return False
+        except_nhs = [nh['ip'] for nh in attr[0]['nexthops']]
+        running_nhs = [nh['ip'] for nh in running_routes[prefix][0]['nexthops']]
+        if except_nhs != running_nhs:
+            return False
+    return True
+
+
+def validate_rx_tx_counters(ptf_dp, end_time, start_time):
```

Comment:
suggest split into 2 methods
1. `caculate_downtime` returns down time only. no side effect and can be re-used in other testcase
2. `validate_downtime` assert down time